### ISSUE TITLE
gradle plugin to run tests that use Spring Framework - Getting Unable to resolve member 

### ISSUE THREAD

**Issue by sguptatw** [manupsunny/gauge-gradle-plugin#20]
**Expected behavior**
Should be able to run from command line, if I am able to run it from the IDE(IntelliJ).

**Actual behavior**
They run successfully in IntelliJ, but fail when run from the command line. 

**Steps to replicate**
* Project [ioc-spring.zip](https://github.com/getgauge/gauge/files/1673217/ioc-spring.zip)
* Run from the IntelliJ IDE. It is successful
* gradle clean gauge
```
Exception in thread "main" org.reflections.ReflectionsException: Can't resolve member named failurePayment for class PaymentService
at org.reflections.util.Utils.getMemberFromDescriptor(Utils.java:94)
at org.reflections.util.Utils.getMethodsFromDescriptors(Utils.java:101)
at org.reflections.Reflections.getMethodsAnnotatedWith(Reflections.java:482)
at com.thoughtworks.gauge.scan.StepsScanner.scan(StepsScanner.java:38)
at com.thoughtworks.gauge.scan.ClasspathScanner.scan(ClasspathScanner.java:39)
at com.thoughtworks.gauge.GaugeRuntime.connectSynchronously(GaugeRuntime.java:82)
at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:54)
[ValidationError] /Users/johncook/temp/ioc-spring/specs/paymentService.spec:4 Connection closed [127.0.0.1:54374] cause: read tcp 127.0.0.1:54371->127.0.0.1:54374: read: connection reset by peer => 'Set debit card balance as "1000" rupees'
[ValidationError] /Users/johncook/temp/ioc-spring/specs/paymentService.spec:8 write tcp 127.0.0.1:54371->127.0.0.1:54374: use of closed network connection => 'Pay amount "500" rupees'
[ValidationError] /Users/johncook/temp/ioc-spring/specs/paymentService.spec:9 write tcp 127.0.0.1:54371->127.0.0.1:54374: use of closed network connection => 'Pay amount "500" rupees'
[ValidationError] /Users/johncook/temp/ioc-spring/specs/paymentService.spec:13 write tcp 127.0.0.1:54371->127.0.0.1:54374: use of closed network connection => 'Fail to pay "2000" rupees'
process /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre/bin/java with pid 4602 quit unexpectedly. exit status 1
Failed to initialize suite datastore. Error: write tcp 127.0.0.1:54371->127.0.0.1:54374: use of closed network connection
Successfully generated html-report to => /Users/johncook/temp/ioc-spring/reports/html-report/index.html
Specifications: 0 executed 0 passed 0 failed 0 skipped
Scenarios: 0 executed 0 passed 0 failed 0 skipped

Total time taken: 20ms

BUILD SUCCESSFUL in 2s
4 actionable tasks: 4 executed`
```
**Version**
```
Gauge version: 0.9.8.nightly-2018-01-24
Commit Hash: ac3d8bc

Plugins
-------
java (0.6.6)
```
Gauge Gradle Plugin v1.4.5

**Comment by sguptatw** [comment]
This is the gradle configuration used by user @johncook-snap40 
https://github.com/getgauge/gauge/files/1673667/build.gradle.txt

**Comment by sky-diogorodrigues** [comment]
Are there any updates on this issue? 
+1

**Comment by beekeep** [comment]
I ran into this issue when using gradle 4.5 and figured out the classpath being using by the plugin was incorrect.  A new directory was added to the build/classes directory in gradle 4 that causes the compiled classes to not be found.  The classes were previously built to build/classes/main, but in gradle 4 this was changed to build/classes/java/main. I have created a pull request to fix this issue (https://github.com/manupsunny/gauge-gradle-plugin/pull/21)
